File: app/src/main/java/pl/adriankremski/coolector/repository/Session.java
package pl.adriankremski.coolector.repository;

import android.content.Context;

import net.grandcentrix.tray.TrayPreferences;

import pl.adriankremski.coolector.Constants;

public class Session extends TrayPreferences implements Constants {

    public Session(Context context) {
        super(context, "SESSION", 1);
    }

    public boolean isLoggedIn() {
        return getString(Constants.PreferencesKey.SESSION_TOKEN, null) != null;
    }

    public String getSessionToken() {
        return getString(PreferencesKey.SESSION_TOKEN, null);
    }

    public Session setSessionToken(String token) {
        put(PreferencesKey.SESSION_TOKEN, token);
        return this;
    }

    public String getSessionKey() {
        return getString(PreferencesKey.SESSION_KEY, null);
    }

    public Session setSessionKey(String key) {
        put(PreferencesKey.SESSION_KEY, key);
        return this;
    }

    @Override
    protected void onCreate(int i) {

    }

    @Override
    protected void onUpgrade(int i, int i1) {

    }
}

File: app/src/main/java/com/noordwind/apps/collectively/data/datasource/Session.kt
package com.noordwind.apps.collectively.data.datasource

import android.content.Context
import net.grandcentrix.tray.TrayPreferences
import com.noordwind.apps.collectively.Constants

class Session(context: Context) : TrayPreferences(context, "SESSION", 1), Constants {

    val isLoggedIn: Boolean
        get() = getString(Constants.PreferencesKey.SESSION_TOKEN, null) != null

    val sessionToken: String?
        get() = getString(Constants.PreferencesKey.SESSION_TOKEN, null)

    fun setSessionToken(token: String?): Session {
        put(Constants.PreferencesKey.SESSION_TOKEN, token)
        return this
    }

    val sessionKey: String?
        get() = getString(Constants.PreferencesKey.SESSION_KEY, null)

    fun setSessionKey(key: String): Session {
        put(Constants.PreferencesKey.SESSION_KEY, key)
        return this
    }

    override fun onCreate(i: Int) {

    }

    override fun onUpgrade(i: Int, i1: Int) {

    }
}
